src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManager.java

```diff
@@ -647,7 +655,9 @@ protected JDAImpl buildInstance(final int shardId) throws LoginException, Interr
 
         jda.asBot().setShardManager(this);
 
-        if (this.eventManager != null)
+        if(this.eventManagerProvider != null)
+        	jda.setEventManager(this.eventManagerProvider.get());
+        else if (this.eventManager != null)
```

Comment:
There is no need to have both of these, you can simply make the setEventManager method create a provider for it.